docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+         * @var UpdatePageInterface
+         */
+        private $updatePage;
+
+        /**
+         * @param IndexPageInterface $indexPage
+         * @param CreatePageInterface $createPage
+         * @param UpdatePageInterface $updatePage
+         */
+        public function __construct(
+            IndexPageInterface $indexPage,
+            CreatePageInterface $createPage,
+            UpdatePageInterface $updatePage
+        ) {
+            $this->indexPage = $indexPage;
+            $this->createPage = $createPage;
+            $this->updatePage = $updatePage;
+        }
+
+        /**
+         * @Given I want to add a new country
+         */
+        public function iWantToAddNewCountry()
+        {
+            $this->createPage->open(); // This method will send request.
+        }
+
+        /**
+         * @When I choose :countryName
+         */
+        public function iChoose($countryName)
+        {
+            $this->createPage->chooseName($countryName);
+            // Great benefit of using page objects is that we hide html manipulation behind a interfaces so we can inject different CreatePage which implements CreatePageInterface
+            // And have different html elements which allows for example chooseName($countryName).
+        }
+
+        /**
+         * @When I add it
+         */
+        public function iAddIt()
+        {
+            $this->createPage->create();
+        }
+
+        /**
+         * @Then /^the (country "([^"]+)") should appear in the store$/
+         */
+        public function countryShouldAppearInTheStore(CountryInterface $country) // This step use Country transformer to get Country object.
+        {
+            $this->indexPage->open();
+
+            //Webmozart assert library.
+            Assert::true(
+                $this->indexPage->isSingleResourceOnPage(['code' => $country->getCode()]),
+                sprintf('Country %s should exist but it does not', $country->getCode())
+            );
+        }
+    }
+
+.. code-block:: php
+    namespace Sylius\Behat\Context\Ui\Admin
+
+    final class NotificationContext implements Context
+    {
+        /**
+         * @var NotificationCheckerInterface
+         */
+        private $notificationChecker; // This is a helper service which give access to proper notification elements.
+
+        /**
+         * @param NotificationCheckerInterface $notificationChecker
+         */
+        public function __construct(NotificationCheckerInterface $notificationChecker)
+        {
+            $this->notificationChecker = $notificationChecker;
+        }
+
+        /**
+         * @Then I should be notified that it has been successfully created
+         */
+        public function iShouldBeNotifiedItHasBeenSuccessfullyCreated()
+        {
+            $this->notificationChecker->checkNotification('has been successfully created.', NotificationType::success());
+        }
+    }
+
+Transformer contexts
+~~~~~~~~~~~~~~~~~~~~
+
```

Comment:
You should also add warning, that no object should be created in transformers. Only allowed exceptions are Customer and Address.